src/com/lftechnology/java/training/srijan/basics/javaio/CharacterDataHandler.java

```diff
@@ -0,0 +1,82 @@
+package com.lftechnology.java.training.srijan.basics.javaio;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CharacterDataHandler {
+	private String inputString;
+	private String inputCharacter;
+	private int count;
+	private static final Logger LOGGER = Logger.getLogger(CharacterDataHandler.class.getName());
+
+	public CharacterDataHandler(String inputString, String inputCharacter) {
+		this.inputString = inputString;
+		this.inputCharacter = inputCharacter;
+		this.count = 0;
+	}
+/**
+ * <p>Handles all the operation of the application</p>
+ * @author srijan
+ */
+	public void dataHandler() {
+		try {
+			displayInputString();
+			writeInFile();
+			readFromFile();
+		} catch (FileNotFoundException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)